libs/cardano-ledger-core/src/Cardano/Ledger/UTxO.hs

```diff
@@ -102,7 +103,7 @@ instance
   where
   type Share (UTxO era) = Interns (Credential 'Staking)
   decShareCBOR credsInterns =
-    UTxO <$!> decodeMap decCBOR (decShareCBOR credsInterns)
+    UTxO <$!> decodeMap decNoShareCBOR (decShareCBOR credsInterns)
```

Comment:
Why this change? Is it just to be more explicit while keeping the same behaviour?

Reply:
This is necessary because the `DecShareCBOR` instance for `TxIn` is capable of decoding MemPack serialized in CBOR TxIn as well as CBOR encoded TxIn.